### PR TITLE
refactor(components): [time-picker] setup migration

### PR DESCRIPTION
**Pull request by jw-foss**
- Migrate panel-month-range to setup

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Review by tolking — COMMENTED** [review]
It might be better to handle namespaces while refactoring

File: packages/components/date-picker/src/date-picker-com/panel-month-range.vue [hunk 1]
```diff
@@ -96,231 +96,233 @@
...
 
-    const onSelect = (selecting) => {
-      rangeState.value.selecting = selecting
-      if (!selecting) {
-        rangeState.value.endDate = null
-      }
-    }
+const isValidValue = (value: [Dayjs | undefined, Dayjs | undefined]) => {
+  return (
+    Array.isArray(value) &&
```
Comment (btea) [inline packages/components/date-picker/src/date-picker-com/panel-month-range.vue:240]:
```suggestion
    isArray(value) &&
```

**Comment by JeremyWuuuuu** [comment]
> refactoring

@iamkun is ding the namespacing thing.

File: packages/components/date-picker/src/date-picker-com/panel-month-range.vue [hunk 2]
```diff
@@ -96,231 +96,233 @@
...
-          const defaultArr = getDefaultValue()
-          leftDate.value = defaultArr[0]
-          rightDate.value = defaultArr[1]
-        }
-      },
-      { immediate: true }
-    )
+const getDefaultValue = () => {
+  let start: Dayjs
+  if (Array.isArray(defaultValue.value)) {
```
[the remark this answers is not in the document]
Comment (JeremyWuuuuu) [reply packages/components/date-picker/src/date-picker-com/panel-month-range.vue]: Updated

File: packages/components/date-picker/src/date-picker-com/panel-month-range.vue (hunk 1 above)
Comment (JeremyWuuuuu) [reply packages/components/date-picker/src/date-picker-com/panel-month-range.vue]: Updated

**Review by btea — APPROVED** [review]

**Review by tolking — APPROVED** [review]

**Pull request merged**